src/onnx/parse_pooling.cpp

```diff
@@ -19,22 +19,27 @@ struct parse_pooling : op_parser<parse_pooling>
         return {{"AveragePool", "average"},
                 {"GlobalAveragePool", "average"},
                 {"GlobalMaxPool", "max"},
-                {"MaxPool", "max"}};
+                {"MaxPool", "max"},
+                {"LpPool", "lpnorm"},
+                {"GlobalLpPool", "lpnorm"}};
     }
 
     instruction_ref parse(const op_desc& opd,
                           const onnx_parser& /*parser*/,
                           onnx_parser::node_info info,
                           std::vector<instruction_ref> args) const
     {
-        std::string mode = opd.op_name;
-        if(mode != "max" && mode != "average")
+        std::vector<std::string> valid_modes = {"max", "average", "lpnorm"};
+        std::string mode                     = opd.op_name;
+        if(std::find(valid_modes.begin(), valid_modes.end(), mode) == valid_modes.end())
         {
-            MIGRAPHX_THROW("onnx pooling mode must be \"max\" or \"average\"");
+            MIGRAPHX_THROW("onnx pooling mode must be [\"max\", \"average\", \"lpnorm\"]");
         }
-        operation op = make_op(
-            "pooling",
-            {{"mode", mode == "average" ? op::pooling_mode::average : op::pooling_mode::max}});
+        const std::unordered_map<std::string, op::pooling_mode> mode_map = {
```

Comment:
You can just search over `mode_map` for valid_modes. That would eliminate need of `valid_modes`.